fix: add uncles and withdrawals to eth_getBlockByNumber and eth_getBlockByHash

morph-dev opened the pull request:
### What was wrong?

The eth_getBlockByNumber and eth_getBlockByHash don't populate uncles and withdrawals fields of the response (but we have data available).

### How was it fixed?

Populate required fields.

**Note:** I don't know how `size` field should be calculated. Let me know if you do.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] Clean up commit history and use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

KolbyML commented:
> Note: I don't know how size field should be calculated. Let me know if you do.

https://github.com/ethereum/go-ethereum/blob/add5709cb5a02183df8541505fc8e0f8c569faea/core/types/block.go#L429-L439

So I look at Geth and apparently, the size is the `len(rlp(header, uncles, transactions, withdrawals))`

I think it makes sense to include the size in this PR as it is supposed to be a required field of this endpoint

morph-dev commented:
Added `size` as well.
Also removed setting `total_difficulty` in Header as current logic is not correct (and I believe we don't have this info available).
Ready for another view.

njgheorghita commented on ethportal-api/src/types/execution/header.rs:
```diff
@@ -277,7 +277,7 @@ impl From<Header> for RpcHeader {
             hash,
             parent_beacon_block_root: parent_beacon_block_root.map(|h264| h264.0.into()),
-            total_difficulty: Some(difficulty),
+            total_difficulty: None,
```
Could you just add a comment here explaining why this is set to `None`

morph-dev replied:
I think they are still part of the accumulators (based on quick look [here](https://github.com/ethereum/trin/blob/4692e6bde873243884f3820a8b417616efca26c9/trin-validation/src/accumulator.rs)), but they are not part of the content value object.

So yes, you are right that we don't have access to them.
If block is post-merge, we could set it as value is now always the same (basically have hardcoded value), but since we don't have it pre-merge, I think it's better to just say it's unsupported.

njgheorghita commented on tests/rpc_server.rs:
```diff
@@ -111,7 +112,7 @@ async fn test_eth_get_block_by_number() {
 
     assert_header(&block.header, &hwp.header);
-    assert_eq!(block.size, None);
+    assert_eq!(block.size, Some(U256::from(37729)));
```
Hmmm... this is for block `17510000`? Seems like there's a different size value on etherscan: https://etherscan.io/block/17510000

morph-dev replied:
After couple hours of debugging I figured out the problem.

The Transactions should be encoded with envelope (additional rlp header for non-legacy transactions). As far as I can tell, there is no easy way to do so with current code, so I'm leaving `size` as `None` for now (but I added comment for future fix).

carver reviewed (approved):
LGTM, minus the size bug

KolbyML reviewed (approved):
> After couple hours of debugging I figured out the problem.
> 
> The Transactions should be encoded with envelope (additional rlp header for non-legacy transactions). As far as I can tell, there is no easy way to do so with current code, so I'm leaving `size` as `None` for now (but I added comment for future fix).

I wrote code to handle this so I will make a PR to add size I guess and we can merge this now
